scala-effect-api/src/main/scala/com/avast/metrics/scalaeffectapi/impl/GaugeFactoryImpl.scala

```diff
@@ -0,0 +1,30 @@
+package com.avast.metrics.scalaeffectapi.impl
+
+import cats.effect.Sync
+import com.avast.metrics.scalaapi.{Monitor => SMonitor}
+import com.avast.metrics.scalaeffectapi.{Gauge, GaugeFactory}
+
+import java.util.concurrent.atomic.{AtomicLong, AtomicReference}
+
+private class GaugeFactoryImpl[F[_]: Sync](monitor: SMonitor) extends GaugeFactory[F] {
+  override def long(name: String): Gauge[F, Long] = new Gauge[F, Long] {
+    private[this] val valueRef = new AtomicLong()
+    private[this] val gauge = monitor.gauge(name)(() => valueRef.get())
```

Comment:
have you considered using [Ref](https://typelevel.org/cats-effect/docs/2.x/concurrency/ref) directly?

Reply:
good point